File: t21/t.cc
#include <iostream>

using namespace std;

class Int;
class C1;
class C2;

class Int
{
public:
    virtual void coolFunc()= delete;
};

class C1
{
public:
    void coolFunc()
    {
        cout << "Hello, World!" << endl;
        return;
    }
};

class C2 : public C1, public Int
{
public:
};

int main(int argc, char* argv[])
{
    C2* c2 =  new C2;
    Int* inte = (Int*)c2;
    C2* c2_2 = (C2*)inte;
    c2_2->coolFunc();
}

File: t21/t.go
package main

import "fmt"

type Int interface {
	CoolFunc()
}

type C1 struct {
}

func (self *C1) CoolFunc() {
	fmt.Println("worked")
	return
}

type C2 struct {
	C1
}

func main() {

	var c Int

	c = new(C2)

	c.CoolFunc()

}
